### PR TITLE
Bump to mono:2018-10

### PR DESCRIPTION
**Pull request by directhex**

### CODE REVIEW

**Comment by rolfbjarne** [comment]
This should probably also include the changes from PR #4596 (mono-2018-08), otherwise you'll just run into the same issues again.

**Comment by baulig** [comment]
@monojenkins build

**Comment by baulig** [comment]
@monojenkins build

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
Device test results:

* Some of the dontlink tvOS tests crash upon start on device due to #5612.
* The dontlink watchOS tests crash in a test in release mode (https://github.com/mono/mono/issues/13006)
* The mono-native-[unified|compat] tests fail on tvOS (https://github.com/xamarin/xamarin-macios/issues/5613)

**Comment by rolfbjarne** [comment]
build

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/57/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/57//console)
✅ Packages: [xamarin.ios-12.7.0.270.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/fa3878391fba56c90271c0125c3bffc0744bf75f/57/package/xamarin.ios-12.7.0.270.pkg) [xamarin.mac-5.7.0.270.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/fa3878391fba56c90271c0125c3bffc0744bf75f/57/package/xamarin.mac-5.7.0.270.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/fa3878391fba56c90271c0125c3bffc0744bf75f/57/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/fa3878391fba56c90271c0125c3bffc0744bf75f/57/jenkins-results/apicomparison/api-diff.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/fa3878391fba56c90271c0125c3bffc0744bf75f/57/jenkins-results/generator-diff/index.html) (no change)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/58/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/58//console)
✅ Packages: [xamarin.ios-12.7.0.276.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/3f512606fb03b262a1dff6e14b200affeea2557c/58/package/xamarin.ios-12.7.0.276.pkg) [xamarin.mac-5.7.0.276.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/3f512606fb03b262a1dff6e14b200affeea2557c/58/package/xamarin.mac-5.7.0.276.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/3f512606fb03b262a1dff6e14b200affeea2557c/58/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/3f512606fb03b262a1dff6e14b200affeea2557c/58/jenkins-results/apicomparison/api-diff.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/3f512606fb03b262a1dff6e14b200affeea2557c/58/jenkins-results/generator-diff/index.html) (no change)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/59/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/59//console)
✅ Packages: [xamarin.ios-12.7.0.287.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/1410090f25b9295dd7662ca8269b06cdb33251fe/59/package/xamarin.ios-12.7.0.287.pkg) [xamarin.mac-5.7.0.287.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/1410090f25b9295dd7662ca8269b06cdb33251fe/59/package/xamarin.mac-5.7.0.287.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/1410090f25b9295dd7662ca8269b06cdb33251fe/59/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/1410090f25b9295dd7662ca8269b06cdb33251fe/59/jenkins-results/apicomparison/api-diff.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/1410090f25b9295dd7662ca8269b06cdb33251fe/59/jenkins-results/generator-diff/index.html) (no change)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Comment by rolfbjarne** [comment]
build

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/60/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/60//console)
✅ Packages: [xamarin.ios-12.7.0.288.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/319de812aa3edc04857d9ee7127c8eef4f33c1e8/60/package/xamarin.ios-12.7.0.288.pkg) [xamarin.mac-5.7.0.288.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/319de812aa3edc04857d9ee7127c8eef4f33c1e8/60/package/xamarin.mac-5.7.0.288.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/319de812aa3edc04857d9ee7127c8eef4f33c1e8/60/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/319de812aa3edc04857d9ee7127c8eef4f33c1e8/60/jenkins-results/apicomparison/api-diff.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/319de812aa3edc04857d9ee7127c8eef4f33c1e8/60/jenkins-results/generator-diff/index.html) (no change)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Comment by dalexsoto** [comment]
Unrelated test failures -> Watch OS error HE0038: Failed to launch the app → [xamarin/maccore#581](https://github.com/xamarin/maccore/issues/581)

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/61/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/61//console)
✅ Packages: [xamarin.ios-12.7.0.296.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/1e9de967899cdd357b69c0f68b4e6ebfa19687e6/61/package/xamarin.ios-12.7.0.296.pkg) [xamarin.mac-5.7.0.296.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/1e9de967899cdd357b69c0f68b4e6ebfa19687e6/61/package/xamarin.mac-5.7.0.296.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/1e9de967899cdd357b69c0f68b4e6ebfa19687e6/61/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/1e9de967899cdd357b69c0f68b4e6ebfa19687e6/61/jenkins-results/apicomparison/api-diff.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/1e9de967899cdd357b69c0f68b4e6ebfa19687e6/61/jenkins-results/generator-diff/index.html) (no change)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Comment by ekene90** [comment]
Good tools

**Comment by VincentDondain** [comment]
# QA mono-integration-58 Report ([link](https://devdiv.visualstudio.com/DevDiv/_releaseProgress?releaseId=264271&_a=release-pipeline-progress))

**TL;DR; Good to merge ✅** 

_Commenting on failing tests to explain why I think they should be considered green (✅)_

## XI / XF sample Tests ✅ 

Those are still running for RD 58 but those succeeded for past builds / only had known issues. Unlikely to be different but I will report back if anything comes up.

## XI Submission Tests ✅ 

Known issue similar to what all other branches are getting: https://github.com/xamarin/QualityAssurance/issues/2990

## XI Smoke Tests ✅ 

Known reporting issue (no device crash logs): https://github.com/xamarin/QualityAssurance/issues/2991

Ran tests locally with and without incremental builds (fastdev) on iPhone X in debug mode (what was failing on bots). Tests all ran fine. We believe this might be an environment issues with devices, will restart them.

## VSM XM HS Smoke Tests ✅ 

They succeeded in previous build and are failing in latest with not much info. I filed an environment issue about it: https://github.com/xamarin/QualityAssurance/issues/2992

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/62/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/62//console)
✅ Packages: [xamarin.ios-12.7.0.301.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/c46a8b9611f93f9fb11e5b2d92fbcfbfdb715087/62/package/xamarin.ios-12.7.0.301.pkg) [xamarin.mac-5.7.0.301.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/c46a8b9611f93f9fb11e5b2d92fbcfbfdb715087/62/package/xamarin.mac-5.7.0.301.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/c46a8b9611f93f9fb11e5b2d92fbcfbfdb715087/62/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/c46a8b9611f93f9fb11e5b2d92fbcfbfdb715087/62/jenkins-results/apicomparison/api-diff.html) (please review changes)
ℹ️ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/c46a8b9611f93f9fb11e5b2d92fbcfbfdb715087/62/jenkins-results/generator-diff/index.html) (please review changes)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-4904/63/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/63/display/redirect) (on internal Jenkins) failed in stage(s) 'Running XM tests on '10.14', Running XM tests on '10.12'' 🔥 : hudson.AbortException: Xamarin.Mac tests on macOS 10.14 failed (xammac_tests)

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/63//console)
✅ Packages: [xamarin.ios-12.7.0.311.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/bbbf15a72bebc85a302d55de3afac389654f6313/63/package/xamarin.ios-12.7.0.311.pkg) [xamarin.mac-5.7.0.311.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/bbbf15a72bebc85a302d55de3afac389654f6313/63/package/xamarin.mac-5.7.0.311.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/bbbf15a72bebc85a302d55de3afac389654f6313/63/jenkins-results/api-diff/index.html)
ℹ️ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/bbbf15a72bebc85a302d55de3afac389654f6313/63/jenkins-results/apicomparison/api-diff.html) (please review changes)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/bbbf15a72bebc85a302d55de3afac389654f6313/63/jenkins-results/generator-diff/index.html) (no change)
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.
🔥 [Xamarin.Mac tests on 10.14 failed: Xamarin.Mac tests on macOS 10.14 failed (xammac_tests)](https://jenkins.internalx.com/job/macios/job/PR-4904/63/display/redirect) 🔥
🔥 [Xamarin.Mac tests on 10.12 failed: Xamarin.Mac tests on macOS 10.12 failed (xammac_tests)](https://jenkins.internalx.com/job/macios/job/PR-4904/63/display/redirect) 🔥

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/64/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/64//console)
✅ Packages: [xamarin.ios-12.7.0.316.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/97b5aef47eab244af24a34d8d8b54e40ddd676f6/64/package/xamarin.ios-12.7.0.316.pkg) [xamarin.mac-5.7.0.316.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/97b5aef47eab244af24a34d8d8b54e40ddd676f6/64/package/xamarin.mac-5.7.0.316.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/97b5aef47eab244af24a34d8d8b54e40ddd676f6/64/jenkins-results/api-diff/index.html)
🔥 [Failed to compare API and create generator diff](https://jenkins.internalx.com/job/macios/job/PR-4904/64//console) 🔥
&nbsp;&nbsp;&nbsp;&nbsp;Failed to update apidiff references
&nbsp;&nbsp;&nbsp;&nbsp;Search for `Comparing API & creating generator diff` in the log to view the complete log.
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.

**Comment by dalexsoto** [comment]
Both test failures and diff issues are known so ✅ 

* API Diff fails when removing the bcl csproj tests. -> [xamarin/maccore#1452](https://github.com/xamarin/maccore/issues/1452)
* Watch OS error HE0038: Failed to launch the app → [xamarin/maccore#581](https://github.com/xamarin/maccore/issues/581)

**Comment by dalexsoto** [comment]
Unrelated test failures

* Watch OS error HE0038: Failed to launch the app → [xamarin/maccore#581](https://github.com/xamarin/maccore/issues/581)

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-4904/65/display/redirect) (on internal Jenkins) succeeded

✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-4904/65//console)
✅ Packages: [xamarin.ios-12.7.0.320.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/221f3e4a375d565d24087afbfcb1d6bc824c1ca9/65/package/xamarin.ios-12.7.0.320.pkg) [xamarin.mac-5.7.0.320.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-4904/221f3e4a375d565d24087afbfcb1d6bc824c1ca9/65/package/xamarin.mac-5.7.0.320.pkg)
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-4904/221f3e4a375d565d24087afbfcb1d6bc824c1ca9/65/jenkins-results/api-diff/index.html)
🔥 [Failed to compare API and create generator diff](https://jenkins.internalx.com/job/macios/job/PR-4904/65//console) 🔥
&nbsp;&nbsp;&nbsp;&nbsp;Failed to update apidiff references
&nbsp;&nbsp;&nbsp;&nbsp;Search for `Comparing API & creating generator diff` in the log to view the complete log.
ℹ️ Test run skipped: Not running tests here because they're run on public Jenkins.
